lib/iris/_lazy_data.py

```diff
@@ -138,7 +124,10 @@ def as_concrete_data(data):
         # In some cases dask may return a scalar numpy.int/numpy.float object
         # rather than a numpy.ndarray object.
         # Recorded in https://github.com/dask/dask/issues/2111.
+        dtype = data.dtype
         data = np.asanyarray(data.compute())
+        if isinstance(data, ma.core.MaskedConstant):
```

Comment:
Again, how is this tied up with the main point ?

Reply:
When I reverted the code to how it was before I found that some of our tests which check that the result of realising lazy data is not a `MaskedConstant` were failing. It seems that Dask is returning a `MaskedConstant` in situations where it wasn't before.